### PR TITLE
Add support for multi-part form data

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Added method WriteMultipartFormData() to Request.

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] Apache v2 license headers are included in each file.
 
[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/master/CHANGELOG.md

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
This was adapted from [track 1](https://github.com/Azure/go-autorest/blob/master/autorest/preparer.go#L306) and has been tested against the local test server.

File: sdk/azcore/request.go [hunk 1]
```diff
@@ -187,6 +188,44 @@ func (req *Request) SetBody(body ReadSeekCloser, contentType string) error {
 	return nil
 }
 
+// WriteMultipartFormData writes the specified keys/values as multi-part form
+// fields with the specified value.  File content must be specified as a ReadSeekCloser.
+// All other values are treated as string values.
+func (req *Request) WriteMultipartFormData(formData map[string]interface{}) error {
```
Comment (jhendrixMSFT) [inline sdk/azcore/request.go:194]: Maybe this should be `SetMultipartFormData()`?  Similar to `SetBody()`

Comment (catalinaperalta) [reply sdk/azcore/request.go:194]: For consistency, I would say Set is better.

**Review by catalinaperalta — APPROVED** [review]
LGTM! I think Set would be better to be consistent with SetBody.

**Review by JeffreyRichter — APPROVED** [review]

File: sdk/azcore/request.go (hunk 1 above)
Comment (jhendrixMSFT) [reply sdk/azcore/request.go]: Agreed, changed.

**Pull request merged**